aws_xray_sdk/ext/httplib/patch.py

```diff
@@ -0,0 +1,126 @@
+import wrapt
+from collections import namedtuple
+
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.models import http
+from aws_xray_sdk.ext.util import inject_trace_header
+
+import ssl
+
+
+_XRAY_PROP = '_xray_prop'
+_XRay_Data = namedtuple('xray_data', ['method', 'host', 'url'])
+
+
+# ? is not a valid entity, and we don't want things after the ? for the segment name
+def _strip_url(url: str):
```

Comment:
Could you move this helper method to `aws_xray_sdk.ext.util`? You use it on requests patch as well. It can be further used to strip any url string on any library patcher.

Reply:
BTW as a helper method a `None` check could be useful in case the caller forgets to do the check.

Reply:
moved, didn't put the None check as that's a fairly low-level API and you'll get an exception of operating on a 'None' already...thoughts?

Reply:
I would suggest make it no-op if input is `None`. This becomes a general helper under a utility module so it could be used on logging, other patching code etc by other developers. It is bad if the SDK throws a None type error from this part.

Also a nitpick could you remove the prefix `_` since its usage is not private?